audiomentations/augmentations/room_simulator.py

```diff
@@ -0,0 +1,362 @@
+from ast import Str
+import sys
+from typing import Optional, Dict
+import numpy as np
+import random
+
+from scipy.signal import convolve
+
+
+from audiomentations.core.transforms_interface import BaseWaveformTransform
+
+
+class RoomSimulator(BaseWaveformTransform):
+    """
+    A ShoeBox Room Simulator. Simulates a cuboid of parametrized size and 
+    average surface absorption coefficient. It also includes a source \
+    and microphones in parametrized locations.
+
+    Use it when you want a ton of synthetic room impulse responses of specific configurations
+    characteristics or simply to quickly add reverb for augmentation purposes
+
+    Some examples:
+
+    > augment = RoomSimulator(use_ray_tracing=False)
+    Quickly add reverberation without caring for accuracy.
+
+    > augment = RoomSimulator(calculate_by_absorption_or_rt60="rt60", min_absorption_value_or_rt60=0.06, max_absorption_value_or_rt60=0.06, min_size_x = ...)
+    Augment with randomly selected room impulse responses that have an RT60 of 0.06.
+
+    > augment = RoomSimulator(min_mic_radius=1.0, max_min_radius=1.0)
+    Augment with a RIR captured by all positions of the microphone on a sphere, centred around the source at 1m
+
+    > augment = RoomSimulator(min_mic_radius=1.0, max_min_radius=1.0, min_mic_elevation=0.0, max_mic_elevation=0.0)
+    Augment with a RIR captured by all positions of the microphone on a circle, centred around the source at 1m    
+    """
+
+    supports_multichannel = True
+
+    def __init__(
+        self,
+        min_size_x: float = 3.6,
+        max_size_x: float = 5.6,
+        min_size_y: float = 3.6,
+        max_size_y: float = 3.9,
+        min_size_z: float = 2.4,
+        max_size_z: float = 3.0,
+        min_absorption_value: float = 0.15,
+        max_absorption_value: float = 0.45,
+        min_target_rt60: float = 0.15,
+        max_target_rt60: float = 0.45,
+        min_source_x: float = 0.1,
+        max_source_x: float = 3.5,
+        min_source_y: float = 0.1,
+        max_source_y: float = 2.7,
+        min_source_z: float = 1.0,
+        max_source_z: float = 2.1,
+        min_mic_distance: float = 0.15,
+        max_mic_distance: float = 0.35,
+        min_mic_azimuth: float = -np.pi,
+        max_mic_azimuth: float = np.pi,
+        min_mic_elevation: float = -np.pi,
+        max_mic_elevation: float = np.pi,
+        calculation_mode: float = "absorption",
+        use_ray_tracing: float = True,
+        max_order: Optional[int] = None,
+        leave_length_unchanged: Optional[bool] = None,
+        padding: float = 0.1,
+        p: float = 0.5,
+        ray_tracing_options: Dict = {
```

Comment:
This default argument is mutable, which means room for problems (no pun intended). Maybe I propose a slightly different way of doing it? https://docs.python-guide.org/writing/gotchas/#what-you-should-do-instead